### PR TITLE
[goto-coverage] fix incorrect assertion coverage

### PR DESCRIPTION
**Pull request by ChenfengWei0**
- issue: #1618 

```c++
#include <assert.h>

void test()
{

    int x = 1;
    for (int i = 0; i < 2; i++)
        assert(x == 1);
}

int test1()
{
    int x = 1;
    for (int i = 0; i < 2; i++)
        assert(x == 1);
}

int main()
{
    test();
    test1();
}
```

- `--goto-coverage --unwind n --no-unwinding-assertion`

```sh
# n = 1
Total Asserts: 2
Reached Assertions Instances: 1
Total Assertion Instances: 4
Assertion Instances Coverage: 25%

# 2
Total Asserts: 2
Reached Assertions Instances: 2
Total Assertion Instances: 4
Assertion Instances Coverage: 50%

# 3
Total Asserts: 2
Reached Assertions Instances: 4
Total Assertion Instances: 4
Assertion Instances Coverage: 100%

# 4
Total Asserts: 2
Reached Assertions Instances: 4
Total Assertion Instances: 4
Assertion Instances Coverage: 100%

# 10
Total Asserts: 2
Reached Assertions Instances: 4
Total Assertion Instances: 4
Assertion Instances Coverage: 100%

# 20
Total Asserts: 2
Reached Assertions Instances: 4
Total Assertion Instances: 4
Assertion Instances Coverage: 100%
```

### CODE REVIEW

**Comment by ChenfengWei0** [comment]
Something weird related to k-induction...
```c++
void loop1()
{
   for (int i = 0; i < 2; i++)
      assert(0);
}

void loop2()
{
   for (int i = 0; i < 4; i++)
      assert(1);
}

int main()

{
   switch (nondet_int())
   {
   case 1:
      loop1();
      break;
   case 2:
      loop2();
      break;
   default:
      break;
   }
   return 0;
}
```
- result
```sh
# --unwind 1 --no-unwinding-assertions
Total Asserts: 2
Reached Assertion Instances: 2
Total Assertion Instances: 6
Assertion Instances Coverage: 33.333333333333336%

# --incremental-bmc
Total Asserts: 2
Reached Assertion Instances: 2
Total Assertion Instances: 6
Assertion Instances Coverage: 33.333333333333336%

# --k-induction
Total Asserts: 2
Reached Assertion Instances: 2
Total Assertion Instances: 2
Assertion Instances Coverage: 100%
```

It might related to issue #1092, where very similar bugs are mentioned

File: src/goto-symex/slice.cpp [hunk 1]
```diff
@@ -176,7 +176,10 @@ bool claim_slicer::run(symex_target_equationt::SSA_stepst &steps)
         claim_to_keep) // this is the assertion that we should not skip!
       {
         it->ignore = false;
-        claim_msg = it->comment;
+        if (!is_goto_cov)
```
Comment (lucasccordeiro) [inline src/goto-symex/slice.cpp:179]: Can you write some comments here?

**Review by lucasccordeiro — APPROVED** [review]
LGTM.

I have just one minor comment.

File: src/goto-symex/slice.cpp (hunk 1 above)
Comment (ChenfengWei0) [reply src/goto-symex/slice.cpp:179]: Added as suggested.

**Comment by ChenfengWei0** [comment]
`--goto-coverage-claims`
```sh
[Coverage]

Total Asserts: 2
Total Assertion Instances: 4
Reached Assertion Instances: 4
  x == 1        file file.c line 8 column 9 function test
  x == 1        file file.c line 8 column 9 function test
  x == 1        file file.c line 15 column 9 function test1
  x == 1        file file.c line 15 column 9 function test1
Assertion Instances Coverage: 100%
```

**Comment by lucasccordeiro** [comment]
Shall we also update the documentation https://ssvlab.github.io/esbmc/documentation.html?

**Comment by ChenfengWei0** [comment]
> Shall we also update the documentation https://ssvlab.github.io/esbmc/documentation.html?

Yeah, I will update the documentation later

**Comment by lucasccordeiro** [comment]
Thanks for submitting this PR, @ChenfengWei0.

**Pull request merged**